### PR TITLE
Encryption Scope SAS

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs [hunk 1]
```diff
@@ -645,6 +645,111 @@ public async Task AppendBlockAsync_EncryptionScope()
...
+            AppendBlobClient blob = InstrumentClient(containerClient.GetAppendBlobClient(GetNewBlobName()));
+
+            // Act
+            Response<BlobContentInfo> response = await blob.CreateIfNotExistsAsync();
+
+            // Assert
+            Assert.AreEqual(TestConfigDefault.EncryptionScope, response.Value.EncryptionScope);
+        }
+
+        [Ignore("TODO, waiting for working test account.")]
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Blobs/tests/AppendBlobClientTests.cs:715]: This is a big TODO.  We need to not merge this until we have seen Identity Encryption Scope SAS work with the service.

**Comment by seanmcc-msft** [comment]
Note that this PR now also includes Data Lake Encryption Scope SAS.

File: sdk/storage/Azure.Storage.Files.DataLake/tests/DirectoryClientTests.cs [hunk 2]
```diff
@@ -350,6 +350,109 @@ public async Task CreateAsync_ConditionsFail()
...
+            DataLakeDirectoryClient sasDirectoryClient = InstrumentClient(new DataLakeDirectoryClient(uriBuilder.ToUri(), GetOptions()));
+
+            // Act
+            Response<PathInfo> response = await sasDirectoryClient.CreateAsync();
+
+            // Assert
+            // TODO
+        }
+
+        [Ignore("TODO need to figure this one out")]
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/tests/DirectoryClientTests.cs:390]: Huge TODO, I haven't gotten Account SAS Encryption Scopes to work with Data Lake yet.

File: sdk/storage/Azure.Storage.Files.DataLake/tests/DirectoryClientTests.cs [hunk 3]
```diff
@@ -350,6 +350,109 @@ public async Task CreateAsync_ConditionsFail()
...
+            DataLakeDirectoryClient directoryClient = InstrumentClient(fileSystemClient.GetDirectoryClient(GetNewDirectoryName()));
+
+            // Act
+            await directoryClient.CreateAsync();
+
+            // Assert
+            // TODO
+        }
+
+        [Ignore("TODO, waiting for working test account.")]
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/tests/DirectoryClientTests.cs:419]: Also need working identity SAS account.

**Review by kasobol-msft — APPROVED** [review]

**Comment by seanmcc-msft** [comment]
Include encryption-scope from clients on GenerateSasUris() and write tests for this case.

**Comment by seanmcc-msft** [comment]
Try calling APIs with x-ms-encryption-scope and an encryption-scope SAS URL and see what happens.

**Comment by seanmcc-msft** [comment]
When soft delete and versioning are enabled together, soft delete protects only deleted versions.
So if you want to list soft deleted entities, you must include versions as well (include=deleted,versions)

**Pull request merged**